doc/source/rllib/rllib-catalogs.rst

```diff
@@ -0,0 +1,178 @@
+       :end-before: __sphinx_doc_end__
+
+The following example showcases how to use the base :py:class:`~ray.rllib.core.models.catalog.Catalog`
+to create an ``encoder`` and an ``action distribution``.
+Besides these, we create a ``head network`` that fits these two by hand to show how you can combine RLLib's
+:py:class:`~ray.rllib.core.models.base.ModelConfig` API and Catalog.
+Extending Catalog to also build this head is how :py:class:`~ray.rllib.core.models.catalog.Catalog` is meant to be
+extended, which we cover later in this guide.
+
+.. dropdown:: **Customize a policy head**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../rllib/examples/catalog/basics/cartpole_models.py
+       :language: python
+       :start-after: __sphinx_doc_begin__
+       :end-before: __sphinx_doc_end__
+
+What are Catalogs
+~~~~~~~~~~~~~~~~~
+
+Catalogs have two primary roles: Choosing the right :py:class:`~ray.rllib.core.models.Model` and choosing the right :py:class:`~ray.rllib.models.distributions.Distribution`.
+By default, all catalogs implement decision trees that decide model architecture based on a combination of input configurations.
+These mainly include the ``observation space`` and ``action space`` of the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`, the ``model config dict`` and the ``deep learning framework backend``.
+
+The following diagram shows the break down of the information flow towards ``models`` and ``distributions`` within an :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`.
+An :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` creates an instance of the Catalog class they receive as part of their constructor.
+It then create its internal ``models`` and ``distributions`` with the help of this Catalog.
+
+.. note::
+    You can also modify :py:class:`~ray.rllib.core.models.Model` or :py:class:`~ray.rllib.models.distributions.Distribution` in an :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` directly by overriding the RLModule's constructor!
+
+.. image:: images/catalog/catalog_and_rlm_diagram.svg
+    :align: center
+
+The following diagram shows a concrete case in more detail.
+
+.. dropdown:: **Example of catalog in a PPORLModule**
+    :animate: fade-in-slide-down
+
+    The :py:class:`~ray.rllib.algorithms.ppo.ppo_catalog.PPOCatalog` is fed an `observation space``, ``action space``,
+    a ``model config dict`` and the ``view requirements`` of the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`.
+    The model config dicts and the view requirements are only of interest in special cases, such as
+    recurrent networks or attention networks. A PPORLModule has four components that are created by the
+    :py:class:`~ray.rllib.algorithms.ppo.ppo_catalog.PPOCatalog`:
+    ``Encoder``, ``value function head``, ``policy head`` and ``action distribution``. You can find out more about this
+    distinction between these components in our section on Models.
```

Comment:
Do you want to link to the section on Models?

Reply:
I'm getting rid of this sentence and will add this info back once the section is added...